client/src/app/members/new-edit/member-new-edit.component.ts

```diff
@@ -0,0 +1,160 @@
+import { CURRENT_SCHOOL_YEAR } from '../../_helpers/currentYear';
+import { ConfirmationDialogComponent } from '../../dialogs/confirmation-dialog/confirmation-dialog.component';
+
+@Component({
+  templateUrl: './member-new-edit.component.html',
+  styleUrls: ['./member-new-edit.component.scss'],
+})
+
+export class MemberNewEditComponent implements OnInit {
+  memberId: number;
+  reRegistration = false;
+  registrationForm: FormGroup;
+  availableSchools = AVAILABLE_SCHOOLS;
+  searchedMembersData: MatTableDataSource<Member>;
+  displayedColumns = ['lastName', 'firstName', 'school', 'lastActive'];
+  searchQuery: string;
+  spinner = false;
+  notEnoughCharacters = false;
+
+  @ViewChild(MatSort) sort: MatSort;
+  @ViewChild(MatPaginator) paginator: MatPaginator;
+
+  constructor(
+    private memberService: MemberService,
+    private toasterService: ToasterService,
+    private router: Router,
+    private formBuilder: FormBuilder,
+    private route: ActivatedRoute,
+    public dialog: MatDialog,
+    ) { }
+
+  ngOnInit(): void {
+    this.initForm();
+    this.initEdit();
+    this.initNameValidation();
+  }
+
+  initForm(): void {
+    this.registrationForm = this.formBuilder.group({
+      lastName: ['', Validators.required],
+      firstName: ['', Validators.required],
+      school: ['', Validators.required],
+      statuts: [false, ValidateCheckbox],
+      ri: [false, ValidateCheckbox],
+    });
+  }
+
+  initNameValidation(): void {
+    this.registrationForm.controls.lastName.valueChanges.subscribe((value) => {
+      this.registrationForm.controls.lastName.setValue(this.nameFormatter(value), { emitEvent: false });
+    });
+    this.registrationForm.controls.firstName.valueChanges.subscribe((value) => {
+      this.registrationForm.controls.firstName.setValue(this.nameFormatter(value), { emitEvent: false });
+    });
+  }
+
+  initEdit(): void {
+    this.route.params.subscribe((params) => {
+      if (params.id) {
+        this.memberId = +params['id'];
+        this.memberService.getById(+this.memberId).subscribe((member) => {
+          this.registrationForm.controls.lastName.setValue(member.lastName);
+          this.registrationForm.controls.firstName.setValue(member.firstName);
+          this.registrationForm.controls.school.setValue(member.school);
+        });
+        this.registrationForm.controls.statuts.setValue(true);
+        this.registrationForm.controls.ri.setValue(true);
+      }
+    });
+  }
+
+  nameFormatter(name: string): string {
+    const newName = !name ? name :
+      name.replace(/[\wÀ-ÿ]+(\S&-)*/g, txt => txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase())
+      .replace(/[`~!@#$%^&*()_|+\=?;:'",.<>\{\}\[\]\\\/]/gi, '');
+    return newName;
+  }
+
+  search(): void {
+    this.searchQuery = this.nameFormatter(this.searchQuery);
+    this.notEnoughCharacters = this.searchQuery.length <= 3 ? true : false;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Simplify this expression. ([typescript:S1125](https://sonarcloud.io/organizations/k-fet/rules#rule_key=typescript%3AS1125))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=k-app&pullRequest=307&issues=AWZCLTrRvOM6zLh9TXFQ&open=AWZCLTrRvOM6zLh9TXFQ)

[ISSUE_KEY]: AWZCLTrRvOM6zLh9TXFQ